cluster-scope/base/rbac.authorization.k8s.io/clusterrolebindings/config-operator-role-binding/clusterrolebinding.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: config-operator-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
```

Comment:
Do you want us to grant you a `cluster-admin` access here? Can you please scope what your service account needs better? I think cluster admin is a bit to generous here.

Reply:
Hi @tumido , the operator need to read/update/create configmap in a namespace. Should I create a role for that ?

Thanks,

Thomas

Reply:
Hey! Ok.. I think we need to take a step back first. What is your intention with this operator? Is this SDAP operator a thing you develop and maintain or is this something you just want to use in your namespace?

If you want to provide this operator as a service to others, you'd need to define a cluster role, sure.

On the other hand, if you want to deploy it just for yourself and keep the operator visibility to your namespace only, you don't need to define any cluster-scoped RBAC. Configmaps are namespaced resources so you can grant this via a role locally within your namespace.

Either way is fine with us, it just needs to be defined before we decide what manifests to use.

I've looked into the sdap sources and they indeed assign `cluster-admin` role, which is quite overkill for any operator in my eyes. We need to narrow the scoped down a bit...

I think you want to go the cluster-scoped operator route (making this operator to work on resources in any namespace on the cluster). If that's the case I think you want a cluster role like this:

```
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: sdap
rules:
  - apiGroups:
      - sdap.apache.org
    resources:
      - gitbasedconfigs
    verbs:
      - create
      - delete
      - deletecollection
      - get
      - list
      - patch
      - update
      - watch
  - apiGroups:
      - ""
    resources:
      - configmaps
    verbs:
      - get
      - list
      - watch
      - create
      - update
      - patch
      - delete
```

I'm unable to find any github repo for nexusjpl/config-operator, can you please point me to the right location? If I can see the actual operator code I can help scope the role better for you. 

If you **don't** want to make this operator available at cluster scope, you can create a namespaced `RoleBinding` resource within your `sdap` namespace and grant the operator any access to your namespace you want - it's your sandboxed playground in the end :slightly_smiling_face: (like even assigning the `admin` `ClusterRole` which grants the operator full project admin access)

Reply:
Thanks @tumido ,
the config-operator is synchronizing git based configuration files  with kubernetes configmap.

Although that might be useful for other we are not planning to support that development for usage outside SDAP. We are barely using it ourselves !

The repo where it is maintained is https://github.com/apache/incubator-sdap-ingester/tree/dev/config_operator

So I think we will go for the option where the operator is only visible to our namespace.

Reply:
Ok, in that case we just need to define aggregate cluster roles so you can access the resource as a namespace admin. All the rest can be set from within your own namespace. :slightly_smiling_face: I'll push an update to your PR with all the necessary manifests.